src/api.js

```diff
@@ -433,3 +433,50 @@ api.declare({
   }
   return res.status(404).send();
 });
+
+api.declare({
+  name: 'createStatus',
+  title: 'Post a status against a given changeset',
+  description: [
+    'For a given changeset (SHA) of a repository, this will attach a "commit status"',
+    'on github. These statuses are links displayed next to each revision.',
+    'The status is either OK (green check) or FAILURE (red cross), ',
+    'made of a custom title and link.',
+  ].join('\n'),
+  stability: 'experimental',
+  method: 'post',
+  // route and query matches github API
+  // https://developer.github.com/v3/repos/statuses/#create-a-status
+  route: '/repository/:owner/:repo/statuses/:sha',
+  input: 'create-status.json',
+  scopes: [['github:create-status:<owner>/<repo>/<sha>']],
```

Comment:
Yes, it may be too precise. Actually, it would prevent posting status for all but lastest changeset of a pull request.
Having said that, it isn't clear how I determine on which branch the current task is running against? And it is even more mysterious how could I know that before calling `api.deplare()`?
(As we have to know that to set the `scopes` attribute right here.